python/ray/serve/task_processor.py

```diff
@@ -0,0 +1,248 @@
+        )
+
+        self._app.conf.update(
+            loglevel="info",
+            worker_pool="threads",
+            worker_concurrency=config.adapter_config.worker_concurrency,
+            task_max_retries=config.max_retry,
+            task_default_queue=config.queue_name,
+            task_ignore_result=False,
+            task_acks_late=True,
+            reject_on_worker_lost=True,
+        )
+
+        if config.adapter_config.broker_transport_options is not None:
+            self._app.conf.update(
+                broker_transport_options=config.adapter_config.broker_transport_options,
+            )
+
+        ### TODO(harshit|2025-07-22): add the failed_task_queue_name and unprocessable_task_queue_name business logic here
+
+    def register_task_handle(self, func, name=None):
+        if name:
+            self._app.task(name=name)(func)
+        else:
+            self._app.task(func)
+
+    def enqueue_task(self, task_name, args=None, kwargs=None, **options) -> TaskResult:
+        task_response = self._app.send_task(
+            task_name,
+            args=args,
+            kwargs=kwargs,
+            queue=self._config.queue_name,
+            **options,
+        )
+
+        return TaskResult(
+            id=task_response.id,
+            status=task_response.status,
+            created_at=time.time(),
+            result=task_response.result,
+        )
+
+    def get_task_status(self, task_id) -> Dict[str, Any]:
+        task_details = self._app.AsyncResult(task_id)
+        return {
+            "id": task_details.id,
+            "result": task_details.result,
+            "status": task_details.status,
+        }
+
+    def cancel_task(self, task_id) -> bool:
+        return self._app.AsyncResult(task_id).cancel()
+
+    def get_metrics(self) -> Dict[str, Any]:
+        return self._app.control.inspect().stats()
+
+    def start_consumer(self, **kwargs):
+        """Starts the Celery worker thread."""
+        if self._worker_thread is not None and self._worker_thread.is_alive():
+            logger.info("Celery worker thread is already running.")
+            return
+
+        self._worker_thread = threading.Thread(
+            target=self._app.worker_main,
+            args=(("worker", f"--hostname={self._app.main}"),),
+        )
+        self._worker_thread.start()
+
+        logger.info(f"Celery worker thread started with hostname: {self._app.main}")
+
+    def stop_consumer(self, timeout: float = 10.0):
+        """Signals the Celery worker to shut down and waits for it to terminate."""
+        if self._worker_thread is None or not self._worker_thread.is_alive():
+            logger.info("Celery worker thread is not running.")
+            return
+
+        logger.info("Sending shutdown signal to Celery worker...")
+
+        # Use the worker's hostname for targeted shutdown
+        self._app.control.broadcast(
+            "shutdown", destination=[f"celery@{self._app.main}"]
+        )
```

Comment:
we should really be starting a worker using the direct worker object. that way we can fully control the workers lifecycle and dont have to rely on `self._app.control.broadcast`. Although I am not sure what `self._app.control.broadcast` actually does